### PR TITLE
fix(error): Add context to FabricError and constructor from Pod status

### PR DESCRIPTION
**Pull request by renatoaguimaraes**

### CODE REVIEW

File: pkg/common/errors/fabric_error.go [hunk 1]
```diff
@@ -5,39 +5,74 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
 )
 
+// Context is a key/value map to carrier any additional information.
+type Context map[string]string
```
Comment (portellaa) [inline pkg/common/errors/fabric_error.go:13]: i think we should change this to `map[string]interface{}`

Comment (renatoaguimaraes) [reply pkg/common/errors/fabric_error.go:13]: I've tried to generate but the controller-ge doesn't accept. 

```
(base) renatoguimaraes@Renatos-MacBook-Pro-16 datasource-controller % make generate 
sed: .env: No such file or directory
go: creating new go.mod: module tmp
go: added github.com/fatih/color v1.12.0
go: added github.com/go-logr/logr v0.4.0
go: added github.com/gobuffalo/flect v0.2.3
go: added github.com/gogo/protobuf v1.3.2
go: added github.com/google/go-cmp v0.5.6
go: added github.com/google/gofuzz v1.1.0
go: added github.com/inconshreveable/mousetrap v1.0.0
go: added github.com/json-iterator/go v1.1.10
go: added github.com/mattn/go-colorable v0.1.8
go: added github.com/mattn/go-isatty v0.0.12
go: added github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: added github.com/modern-go/reflect2 v1.0.1
go: added github.com/spf13/cobra v1.1.3
go: added github.com/spf13/pflag v1.0.5
go: added golang.org/x/mod v0.4.2
go: added golang.org/x/net v0.0.0-20210428140749-89ef3d95e781
go: added golang.org/x/sys v0.0.0-20210510120138-977fb7262007
go: added golang.org/x/text v0.3.6
go: added golang.org/x/tools v0.1.3
go: added golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: added gopkg.in/inf.v0 v0.9.1
go: added gopkg.in/yaml.v2 v2.4.0
go: added gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: added k8s.io/api v0.21.2
go: added k8s.io/apiextensions-apiserver v0.21.2
go: added k8s.io/apimachinery v0.21.2
go: added k8s.io/klog/v2 v2.8.0
go: added k8s.io/utils v0.0.0-20201110183641-67b214c5f920
go: added sigs.k8s.io/controller-tools v0.6.1
go: added sigs.k8s.io/structured-merge-diff/v4 v4.1.0
go: added sigs.k8s.io/yaml v1.2.0
/Users/renatoguimaraes/go/bin/controller-gen object paths=./pkg/models/kube/v2alpha1
github.com/ydataai/datasource-controller/pkg/models/kube/v2alpha1:-: invalid map value type: interface{}
Error: not all generators ran successfully
run `controller-gen object paths=./pkg/models/kube/v2alpha1 -w` to see all available markers, or `controller-gen object paths=./pkg/models/kube/v2alpha1 -h` for usage
make: *** [generate] Error 1
```

Comment (portellaa) [reply pkg/common/errors/fabric_error.go:13]: 🤔  but that will make this not usable for everything i'm afraid. can you see another solution?

Comment (renatoaguimaraes) [reply pkg/common/errors/fabric_error.go:13]: I'm trying to figure out, yesterday I slept on the laptop 😄

**Review by portellaa — APPROVED** [review]

**Pull request merged**